src/super_gradients/common/environment/checkpoints_dir_utils.py

```diff
@@ -58,25 +58,32 @@ def get_checkpoints_dir_path(experiment_name: str, ckpt_root_dir: str = None) ->
     return os.path.join(ckpt_root_dir, experiment_name)
 
 
-def get_ckpt_local_path(source_ckpt_folder_name: str, experiment_name: str, ckpt_name: str, external_checkpoint_path: str):
+def get_ckpt_local_path(source_ckpt_folder_name: str, experiment_name: str, ckpt_name: str, external_checkpoint_path: str, ckpt_root_dir: str = None) -> str:
```

Comment:
Adding to my previous comment, you can see that `get_ckpt_local_path` is called only in 1 place which is the Trainer with `self. source_ckpt_folder_name=None`. So I think we could just remove this `source_ckpt_folder_name` parameter and go with the parameter you introduced which is more explicit.

Besides, the name `source_ckpt_folder_name` was missleading because looking at the code, it was expected to be the local name of the folder including experiments and not the full path

Reply:
removed (including from weight averaging) in latest commit.